multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/MachineTokenRegistry.java

```diff
@@ -42,7 +41,7 @@
    * @param workspaceId id of workspace to generate token for
    * @return generated token value
    */
-  public String generateToken(String userId, String workspaceId) {
+  private String generateToken(String userId, String workspaceId) {
```

Comment:
I'd remove `generateToken` method at all and rewrite `getOrCreateToken` in the following way:
```
  public String getOrCreateToken(String userId, String workspaceId) {
    lock.writeLock().lock();
    try {
      final Map<String, String> wsRow = tokens.row(workspaceId);
      String token = wsRow.get(userId);

      if (token == null) {
        token = generate("machine", 128);
        tokens.put(workspaceId, userId, token);
      }

      return token;
    } finally {
      lock.writeLock().unlock();
    }
```
Why is it better?
1. Write lock is not locked twice.
2. `wsRow.get(userId)` is invoked once instead of two times.

Reply:
ok, good point